commands/start-container.ts

```diff
@@ -41,29 +39,23 @@ export async function startContainerCore(actionContext: IActionContext, context:
     }
 
     if (imageToStart) {
-        docker.getExposedPorts(imageToStart.Id).then((ports: string[]) => {
-            let options = `--rm ${interactive ? '-it' : '-d'}`;
-            if (ports.length) {
-                const portMappings = ports.map((port) => `-p ${port.split("/")[0]}:${port}`); //'port' is of the form number/protocol, eg. 8080/udp.
-                // In the command, the host port has just the number (mentioned in the EXPOSE step), while the destination port can specify the protocol too
-                options += ` ${portMappings.join(' ')}`;
-            }
+        let ports: string[] = [];
+        try {
+            ports = await docker.getExposedPorts(imageToStart.Id);
+        } catch (error) {
+            vscode.window.showWarningMessage(`Unable to retrieve exposed ports: ${parseError(error).message}`);
```

Comment:
So you're intentionally letting the user continue here, correct? That's why it's just `showWarningMessage` without an `await`?

Reply:
Yes.  Mixed feelings about this, though.